liquibase-core/src/main/java/liquibase/database/core/MSSQLDatabase.java

```diff
@@ -349,8 +349,10 @@ public boolean isCaseSensitive() {
         if (caseSensitive == null) {
             try {
                 if (getConnection() != null) {
-                    String collation = ExecutorService.getInstance().getExecutor(this).queryForObject(new RawSqlStatement("SELECT CONVERT(varchar(100), SERVERPROPERTY('COLLATION'))"), String.class);
-                    caseSensitive = ! collation.contains("_CI_");
+                  String catalog = getConnection().getCatalog();
+                  String sql = String.format("SELECT CONVERT(varchar(100), DATABASEPROPERTYEX('%s', 'COLLATION'))", catalog);
```

Comment:
Since the goal of Liquibase is to run arbitrary SQL I tend to not worry too much about sql injection problems so I think it is fine.